ChangeSet
  1.1547 04/04/01 14:28:26 dgibson@wocket.ozlabs.ibm.com +1 -0
  Fix bug in PPC64 hugepage support.  Due to a misunderstanding of
  pmd_offset we could start dereferencing bogus pmd pages.

  arch/ppc64/mm/hugetlbpage.c
    1.12 04/04/01 14:27:57 dgibson@wocket.ozlabs.ibm.com +3 -0
    Fix bug in PPC64 hugepage support.  Due to a misunderstanding of
    pmd_offset we could start dereferencing bogus pmd pages.

diff -Nru a/arch/ppc64/mm/hugetlbpage.c b/arch/ppc64/mm/hugetlbpage.c
--- a/arch/ppc64/mm/hugetlbpage.c	Thu Apr  1 09:18:28 2004
+++ b/arch/ppc64/mm/hugetlbpage.c	Thu Apr  1 09:18:28 2004
@@ -190,6 +190,9 @@
 	BUG_ON(!in_hugepage_area(mm->context, addr));
 
 	pgd = pgd_offset(mm, addr);
+	if (pgd_none(*pgd))
+		return NULL;
+
 	pmd = pmd_offset(pgd, addr);
 
 	/* We shouldn't find a (normal) PTE page pointer here */
.........................................................................
# vim: syntax=diff

